src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls

```diff
@@ -0,0 +1,277 @@
+            BDI_MigrationMappingUtility.updateCustomSettings(BDI_MigrationMappingUtility.HELP_TEXT,
+                e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * @description Save the metadata deployment id in data import custom settings
+     * @param deploymentId Metadata deployment id
+     */
+    @future
+    public static void saveDeploymentId(String deploymentId) {
+        Data_Import_Settings__c dataImportSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
+        dataImportSettings.CMT_Deployment_ID__c = deploymentId;
+
+        try {
+            upsert dataImportSettings;
+        } catch (Exception e) {
+            ERR_Handler.processError(e, ERR_Handler_API.Context.STTG);
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, e.getMessage()));
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Action Method used by Poller to check status of CMDT deployment
+    * @return void
+    */
+    public PageReference startPolling() {
+        timeoutCount++;
+
+        deploymentResult = checkDeploymentStatus();
+        dataImportSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
+
+        if (deploymentResult != null) {
+            // We had a migration that contained CMT records to insert
+            Boolean isSuccess = deploymentResult.status == Metadata.DeployStatus.Succeeded
+                && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING;
+
+            if (isSuccess) {
+                isPolling = false;
+                throwMessage(ApexPages.Severity.CONFIRM, Label.stgDeploymentCustomMetadataMappingEnabled);
+            } else if (deploymentResult.status == Metadata.DeployStatus.FAILED) {
+                isPolling = false;
+                throwMessage(ApexPages.Severity.ERROR, Label.stgDeploymentTryAgain);
+            }
+
+        } else if (deploymentResult == null
+                   && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING) {
+            // We had a migration that didn't contain CMT records to insert
+            // DeploymentContainer was empty which means we're using our standard field mapping set
+            // Org had no conflicting field mappings nor any net new field mappings in their Data Import help text
+            isPolling = false;
+            throwMessage(ApexPages.Severity.CONFIRM, Label.stgDeploymentCustomMetadataMappingEnabled);
+        }
+
+        if (timeoutCount > MAX_TIME_OUT) {
+            // We timed out, reset instance of migration mapping utility
+            migrationMappingUtility = new BDI_MigrationMappingUtility(new BDI_MigrationMappingHelper());
+            isPolling = false;
+            throwMessage(ApexPages.Severity.WARNING, Label.stgDeploymentTimedOut + ' ' + Label.stgDeploymentTryAgain);
```

Comment:
What do you think about making the `throwMessage` method static so another instance of the mapping utility does not need to be instantiated here?

Reply:
I think we can just toss out the line instantiating a new instance of migration util here. It's not actually useful anymore. Left over from before I was clearing the lists holding pending metadata records on every attempted migration.

Reply:
👍🏽